climada/hazard/tc_clim_change.py

```diff
@@ -1,174 +1,329 @@
+        }
+
+    base_start_year, base_end_year = baseline
+    gmst_data, gmst_start_year, gmst_end_year, rcps = get_gmst_info()
+    knutson_data = get_knutson_data()
+
+    nrcps, gmst_years = gmst_data.shape
+
+    if ((base_start_year <= gmst_start_year) or (base_start_year >= gmst_end_year) or
+        (base_end_year <= gmst_start_year) or (base_end_year >= gmst_end_year)):
+
+        raise ValueError("The selected historical baseline falls outside"
+                         "the GMST data period 1880-2100")
+
+    # calculate beta and annual values from this knutson_value
+    # (these annual values correspond to y in the paper)
+
+    var_id = MAP_VARS_NAMES[variable]
+
+    try:
+        basin_id = MAP_BASINS_NAMES[basin]
+        knutson_value = knutson_data[var_id, basin_id, pct]
+
+    except KeyError:
+        # no scaling factors are defined for this basin. Most likely SA.
+        knutson_value = 1
+
+    beta = 0.5 * log(0.01 * knutson_value + 1)
+    tc_properties = np.empty((nrcps, gmst_years))
+
+    for i in range(nrcps):
+        for j in range(gmst_years):
+            tc_properties[i, j] = exp(beta * gmst_data[i, j])
 
-from climada.util.constants import SYSTEM_DIR
+    # calculate baselines for each RCP as averages of the annual values
 
-TOT_RADIATIVE_FORCE = SYSTEM_DIR.joinpath('rcp_db.xls')
-"""© RCP Database (Version 2.0.5) http://www.iiasa.ac.at/web-apps/tnt/RcpDb.
-generated: 2018-07-04 10:47:59."""
+    base_start_index = base_start_year - gmst_start_year
+    base_end_index = base_end_year - gmst_start_year
 
-def get_knutson_criterion():
+    baseline_properties = np.empty(nrcps)
+    for i in range(nrcps):
+        baseline_properties[i] = mean(
+            tc_properties[i, base_start_index:base_end_index + 1]
+            )
+
+    # loop over decades and calculate predicted_properties and fractional changes
+    # (the last decade hits the end of the SST series so is calculated differently)
+
+    mid_years = np.empty(windows_props['windows'])
+    predicted_property_pcs = np.empty((windows_props['windows'], nrcps))
+
+    count = 0
+    for window in range(windows_props['windows']):
+        mid_year = windows_props['start'] + (window) * windows_props['interval']
+        mid_years[window] = mid_year
+        mid_index = mid_year - gmst_start_year
+        actual_smoothing = min(
+            windows_props['smoothing'],
+            gmst_years - mid_index - 1,
+            mid_index
+        )
+        for i in range(nrcps):
+            ind1 = mid_index - actual_smoothing
+            ind2 = mid_index + actual_smoothing + 1
+            prediction = mean(tc_properties[i, ind1:ind2])
+            predicted_property_pcs[count, i] = 100 * (
+                prediction - baseline_properties[i]
+                ) / baseline_properties[i]
+        count += 1
+
+    return pd.DataFrame(predicted_property_pcs,
+                        index=mid_years, columns=rcps)
+
+def get_gmst_info():
```

Comment:
Would it be clearer to the user if this function rather returned a pd.DataFrame with columns 'year', 'rcp', 'gsmt'? Or would this make the calculations more cumbersome later on?